dashboard/rules.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import absolute_import
+
+import rules
+from dashboard.models import User
+
+@rules.predicate
+def is_admin(user):
+    return user.is_superuser
+
+@rules.predicate
+def is_enrolled_in_course(user, course):
+    try:
+        User.objects.get(
+            sis_name=user.get_username(),
+            course_id=course.id,
+        )
+        return True
+    except User.DoesNotExist:
+        return False
+
+@rules.predicate
+def is_instructor_in_course(user, course):
+    try:
+        User.objects.get(
+            sis_name=user.get_username(),
+            course_id=course.id,
+            enrollment_type=User.ENROLLMENT_TYPES.TeacherEnrollment,
+        )
+        return True
+    except User.DoesNotExist:
+        return False
+
+@rules.predicate
+def is_above_student_in_course(user, course):
+    try:
+        enrolled = User.objects.get(
+            sis_name=user.get_username(),
+            course_id=course.id,
+            enrollment_type__in=[
+                User.ENROLLMENT_TYPES.TeacherEnrollment,
+                User.ENROLLMENT_TYPES.TaEnrollment,
+            ],
+        )
+        return True
+    except User.DoesNotExist:
+        return False
+
+
+is_admin_or_enrolled_in_course = is_admin | is_enrolled_in_course
+
+# api
+rules.add_perm('dashboard.get_course_info', is_admin_or_enrolled_in_course)
+rules.add_perm('dashboard.file_access_within_week', is_admin_or_enrolled_in_course)
+rules.add_perm('dashboard.grade_distribution', is_admin_or_enrolled_in_course)
+rules.add_perm('dashboard.update_user_default_selection_for_views', is_admin_or_enrolled_in_course)
+rules.add_perm('dashboard.get_user_default_selection', is_admin_or_enrolled_in_course)
+rules.add_perm('dashboard.assignments', is_admin_or_enrolled_in_course)
```

Comment:
The rules feature looks fine to me but I couldn't actually test this locally due to load problem error. One concern for me is these rules provide a way to preventing users from accessing other course content. In that case, this PR is not handling it, If not this PR at least another issue must be created stating to handle it from UI, like showing "Permission denied" . I know with rewrite of UI in react it would be a duplicate effort for handling it.